### PR TITLE
Ignore non-string property values for trackEvent and trackPage in Ana…

### PR DESCRIPTION
**Pull request by jaelim-ms**
…lytics

### CODE REVIEW

File: MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m [hunk 1]
```diff
@@ -136,30 +136,50 @@ + (BOOL)isAutoPageTrackingEnabled {
...
+- (void)trackPage:(NSString *)pageName withProperties:(NSDictionary<NSString *, NSString *> *)properties {
   if (![super isEnabled])
     return;
 
   // Create and set properties of the event log.
   MSPageLog *log = [[MSPageLog alloc] init];
   log.name = pageName;
-  if (properties)
+  if (properties) {
+    // Check if property dictionary contains non-string values.
```
Comment (clpolet) [inline MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m]: Should add a blanc line before this comment too :)

File: MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m [hunk 2]
```diff
@@ -136,30 +136,50 @@ + (BOOL)isAutoPageTrackingEnabled {
...
   if (![self isEnabled])
     return;
 
   // Create and set properties of the event log.
   MSEventLog *log = [[MSEventLog alloc] init];
   log.name = eventName;
   log.eventId = MS_UUID_STRING;
-  if (properties)
+  if (properties) {
+    // Check if property dictionary contains non-string values.
```
Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m]: Done

File: MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m (hunk 1 above)
Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/MSAnalytics.m]: Done

File: MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h [hunk 3]
```diff
@@ -9,33 +9,40 @@
...
+ * @return YES if page tracking is enabled and NO if disabled.
  */
 + (BOOL)isAutoPageTrackingEnabled;
 
+/**
+ * Validate value types of properties.
+ *
+ * @return YES if properties have valid value types, NO otherwise.
+ */
+- (BOOL)validatePropertyValueType:(NSDictionary<NSString *, NSString *> *)properties;
```
Comment (jaelim-ms) [inline MobileCenterAnalytics/MobileCenterAnalytics/Internals/MSAnalyticsInternal.h]: Added.

**Pull request merged**